### ISSUE TITLE
[bug] Wrong array.mapi and array.map implementation

### ISSUE THREAD

**Issue by nlchar** [cqfn/eo#195]
The implementation of `array.mapi` misses `.copy` method call. Because of it the `array` becomes mutable. Because of it `mapi` and `map` attribute objects are unsafe and cannot be used at all (because EO runtime evaluates the `array` multiple times and on the second iteration the objects are already mapped and do not contain the attributes the objects previously had).
```JAVA
/**
 * MAPI.
 *
 * @since 1.0
 */
public class EOarray$EOmapi extends PhDefault {

    public EOarray$EOmapi(final Phi parent) {
        super(parent);
        this.add("f", new AtFree());
        this.add("φ", new AtBound(new AtLambda(this, self -> {
            final Phi[] array = new Data.Take(
                self.attr("ρ").get() /* HERE */
            ).take(Phi[].class);
            for (int idx = 0; idx < array.length; ++idx) {
                final Phi after = self.attr("f").get().copy();
                after.attr(0).put(array[idx]);
                after.attr(1).put(new Data.Value<>(idx));
                array[idx] = after;
            }
            return new Data.ToPhi(array);
        })));
    }

}

```
Hence, this example cannot be run:
```
+package sandbox
+alias sprintf org.eolang.txt.sprintf
+alias stdout org.eolang.io.stdout

[args...] > app
  map. > squares!
    args
    [current]
      pow. > @
        current.toInt
        2

  stdout > @
    sprintf
      "%d\n"
      reduce.
        squares
        0
        [accumulator current]
          accumulator.add current > @

```

**Comment by yegor256** [comment]
@nlchar good point. Can you make a PR?

**Comment by nlchar** [comment]
@yegor256 yes I'll fix all the issues
#192 
#193 
#194 
#195
and
#186 

The only problem is how to fix #191

**Comment by 0crat** [comment]
You @nlchar[/z](https://www.0crat.com/u/nlchar) need to have one of these roles in order to do "Request order start": `ARC`, `PO`; I'm sorry to say this, but at the moment you've got no roles in this project
